### PR TITLE
Remove guzzle dependency

### PR DESCRIPTION
**Pull request by smoucka**
Swap out guzzle for ext-curl. Have not run unit tests.

### CODE REVIEW

File: src/KlaviyoAPI.php [hunk 1]
```diff
@@ -156,18 +152,25 @@ private function request( $method, $path, $options, $isPublic = false, $isV1 = f
     {
         $options = $this->prepareAuthentication( $options, $isPublic, $isV1 );
 
-        $response = $this->client->request( $method, $path, $options );
+        $setopt_array = $this->getDefaultCurlOptions($method);
```
Comment (remstone7) [inline src/KlaviyoAPI.php:155]: could we chain this so we don't set the same variable 3 times in a row?

Comment (smoucka) [reply src/KlaviyoAPI.php:155]: Yea, was going to go for something like this:
```
        $setopt_array = array_merge(
            $this->getDefaultCurlOptions($method),
            $this->getCurlOptUrl($path, $options),
            $this->getSpecificCurlOptions($options)
        );
```
However I realized merging arrays that use numeric keys (all those curl constants) for an associative array get reindexed :(. We can combine them and preserve their keys using "+"

Wish we had a nice python update method!

**Comment by remstone7** [comment]
this looks good but tests are broken as we use guzzle in them

File: tests/KlaviyoAPITest.php [hunk 2]
```diff
@@ -271,4 +258,121 @@ public function returnCheckProfile( $profiles )
...
+            ),
+            Klaviyo::JSON => array(
+                'list_name' => 'refactor test'
+            )
+        );
+
+        $expected = array(
+            CURLOPT_HTTPHEADER => array(
+                Klaviyo::API_KEY_HEADER . ': ' . $this->testPrivateKey,
+                Klaviyo::USER_AGENT . ': Klaviyo-PHP/' . Klaviyo::PACKAGE_VERSION,
```
Comment (remstone7) [inline tests/KlaviyoAPITest.php:324]: interesting, i take it as this is how curl headers return.

**Pull request merged**